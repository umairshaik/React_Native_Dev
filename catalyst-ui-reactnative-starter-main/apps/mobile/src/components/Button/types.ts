import {ReactElement} from 'react';
import {PressableProps, StyleProp} from 'react-native';

export type ButtonType = 'filled' | 'outlined' | 'text';

export type ButtonProps = PressableProps & {
  text?: string;
  icon?: ReactElement;
  type?: ButtonType;
  textStyle?: StyleProp<Text>;
  accessibleProps: string;
};
