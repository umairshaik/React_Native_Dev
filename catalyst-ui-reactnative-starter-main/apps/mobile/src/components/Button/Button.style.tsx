import {Pressable, Text} from 'react-native';
import styled from 'styled-components/native';
import {ButtonType} from './types';

type Button = {
  type: ButtonType;
};

export const CatalystButton = styled(Pressable)<Button>`
  border-radius: 5px;
  padding: 12px 15px;
  ${({type, theme}) =>
    type === 'filled' && `background-color: ${theme.colors.primary};`}
  ${({type, theme}) =>
    type === 'outlined' &&
    `
    background-color: transparent;
    border-color: ${theme.colors.secondary};
    border-width: 1px;
    color: ${theme.colors.primary}
    `}
`;
export const ButtonText = styled(Text)<Button>`
  font-weight: ${props => props.theme.fonts.titleMedium.fontWeight};
  font-size: ${props => props.theme.fonts.titleMedium.fontSize}px;
  line-height: ${props => props.theme.fonts.titleMedium.lineHeight}px;
  align-self: center;
  ${({type, theme}) => type === 'filled' && `color: ${theme.colors.onPrimary};`}
  ${({type, theme}) => type === 'outlined' && `color: ${theme.colors.primary}`}
`;
