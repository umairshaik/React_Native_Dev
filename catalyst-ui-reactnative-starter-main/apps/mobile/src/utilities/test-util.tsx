import {FC, ReactElement, ReactNode} from 'react';
import {render, RenderAPI} from '@testing-library/react-native';
import {QueryClient, QueryClientProvider} from 'react-query';
import {ThemeProvider} from '../context/theme/themeContext';

const query = new QueryClient({
  defaultOptions: {
    queries: {
      retry: false,
      cacheTime: 0,
    },
  },
});
afterEach(() => {
  query.clear();
});
const AllTheProviders: FC<{children: ReactNode}> = ({children}) => (
  <ThemeProvider>
    <QueryClientProvider client={query}>{children}</QueryClientProvider>
  </ThemeProvider>
);

const customRender = (ui: ReactElement, options?: Omit<RenderAPI, 'wrapper'>) =>
  render(ui, {wrapper: AllTheProviders, ...options});

export * from '@testing-library/react-native';
export {customRender as render};
