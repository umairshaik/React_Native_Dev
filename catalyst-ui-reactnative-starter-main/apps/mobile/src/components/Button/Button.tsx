import React from 'react';
import {CatalystButton, ButtonText} from './Button.style';
import {ButtonProps} from './types';

const Button = ({
  text,
  style,
  accessibleProps,
  textStyle,
  onPress,
  icon,
  type = 'filled',
}: ButtonProps) => (
  <CatalystButton
    onPress={onPress}
    type={type}
    style={style}
    testID={accessibleProps}
    accessibilityHint={`click to ${accessibleProps}`}
    accessibilityValue={{
      text: accessibleProps,
    }}>
    <ButtonText style={textStyle || {}} type={type}>
      {icon}
      {text}
    </ButtonText>
  </CatalystButton>
);
export default Button;
