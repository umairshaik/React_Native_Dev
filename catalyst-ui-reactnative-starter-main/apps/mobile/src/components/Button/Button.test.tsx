import Button from './Button';
import {act, fireEvent, render, screen} from '../../utilities/test-util';

describe('[Component] - [Button]', () => {
  test('should render button component correctly', () => {
    const button = render(
      <Button
        text="Press me"
        onPress={jest.fn()}
        accessibleProps="mock-button"
      />,
    );
    expect(button).toMatchSnapshot();
  });

  test('should be clickable', () => {
    const mockFunction = jest.fn();
    render(
      <Button
        text="Press me"
        onPress={mockFunction}
        accessibleProps="mock-button"
      />,
    );
    act(() => {
      fireEvent.press(screen.getByText('Press me'));
    });

    expect(mockFunction).toBeCalledTimes(1);
  });
});
