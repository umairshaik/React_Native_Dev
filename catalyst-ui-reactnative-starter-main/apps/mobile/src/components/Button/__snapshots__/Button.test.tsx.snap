// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] - [Button] should render button component correctly 1`] = `
<View
  accessibilityHint="click to mock-button"
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": "mock-button",
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "backgroundColor": "#6750A4",
        "borderRadius": 5,
        "paddingBottom": 12,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 12,
      },
    ]
  }
  testID="mock-button"
  type="filled"
>
  <Text
    style={
      [
        {
          "alignSelf": "center",
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontWeight": "500",
          "lineHeight": 24,
        },
        {},
      ]
    }
    type="filled"
  >
    Press me
  </Text>
</View>
`;
